n3fit/src/evolven3fit_new/utils.py

```diff
@@ -0,0 +1,180 @@
+from scipy.interpolate import interp1d
+import numpy as np
+import math
+from reportengine.compat import yaml
+import shutil
+from validphys.pdfbases import PIDS_DICT
+import pathlib
+
+
+class LhapdfLike:
+    """
+    Class which emulates lhapdf but only for an initial condition PDF (i.e. with only one q2 value)
+    """
+
+    def __init__(self, pdf_grid, q20, x_grid):
+        self.pdf_grid = pdf_grid
+        self.q20 = q20
+        self.x_grid = x_grid
+        self.funcs = [
+            interp1d(self.x_grid, self.pdf_grid[pid], kind="cubic")
+            for pid in range(len(PIDS_DICT))
+        ]
+
+    def xfxQ2(self, pid, x, q2):
+        """Return the value of the PDF for the requested pid and x value. If the requested q2
+        value is different from the (only) value available, it raises an error.
+
+        Parameters
+        ----------
+
+            pid: int
+                pid index of particle
+            x: float
+                x-value
+            q2: float
+                Q square value
+
+        Returns
+        -------
+            : float
+            x * PDF value
+        """
+        if not math.isclose(q2, self.q20, rel_tol=1e-6):
```

Comment:
```suggestion
        if not math.isclose(q2, self.q20, rel_tol=1e-6):
```

I would use `np` for this check (not that it matters, it's just one less import)